docs/API/defineFetch-share.md

```diff
@@ -0,0 +1,113 @@
+# `defineFetch.share` API
+
+The `share` property on 'defineFetch' allows other `defineFetch` actions to store data on the same key.
+
+A common use case for `share` are CRUD endpoints that need to update the same data store key through `get` and `put` http requests.
+
+### `share Props`
+
+---
+
+| Prop        | Type             | Default    | Description                                                                                                        |
+| ----------- | ---------------- | ---------- | ------------------------------------------------------------------------------------------------------------------ |
+| `namespace` | string           | required\* | Another `defineFetch` function sharing the same name space will result them sharing the same key to the data store |
+| `merge`     | func(prev, next) | required\* | `merge` adds customizing functionality before the data is applied to the data store.                               |
+
+&nbsp;
+
+Example:
+
+```ts
+// component/actions/getPerson.ts
+
+import defineFetch from '@sift/resift/defineFetch';
+
+export default defineFetch({
+  displayName: 'Get Person',
+  share: {
+    namespace: 'person',
+    // Keep data flow as is
+    merge: (previous, next) => ({ ...previous, ...next }),
+  },
+  make: (personName: string) => ({
+    key: [personName],
+    request: () => ({ http }) =>
+      return http({
+        method: 'GET',
+        route: `/person/${personId}`,
+      }),
+  }),
+});
+```
+
+```ts
+// component/actions/editPerson.ts
+
+import defineFetch from '@sift/resift/defineFetch';
+
+export default defineFetch({
+  displayName: 'Edit Person',
+  share: {
+    namespace: 'person',
+    merge: (previous, next) => {
+      // Customize and add data if needed
+      return { ...previous, ...next, lastEdited: new Date() }
+    },
+  },
+  make: (personName: string) => ({
+    key: [personName],
+    request: (personData: any) => ({ http }) =>
+      return http({
+        method: 'PUT',
+        route: `/person/${personId}`,
+        data: personData
+      }),
+  }),
+});
+```
+
+```tsx
+// Example container using `definefetch` above
+// - component/index.tsx
+
+import React, { useEffect, useCallback } from 'react';
+
+// Required Resift tools
+import useFetch from '@sift/resift/useFetch';
+import useDispatch from '@sift/resift/useDispatch';
+
+// Returned Component
+import Profile from './Profile';
+
+// Action in example above
+import makeGetPerson from './actions/getPerson';
```

Comment:
@ricokahler, I'm a huge fan of... 
- Get instead of fetch
- Update instead of put
- Create instead of post
- delete === delete

Reply:
I agree with you on those names but I think you misunderstood my comment. "get" doesn't also mean "fetch". I think it would be beneficial to always postfix with `-fetch` and additionally prefix with `make-` (depending on whether or not the fetch is a fetch factory or a fetch instance and if the CRUD operation is "get" then I think we can omit the operation but that's also debatable.

For example, if the fetch updated the existing person, I would name the file `makeUpdatePersonFetch.js`. I'm not super happy with this naming convention because of the length but what I do like about it is that the conventions make it clear what is a fetch factory vs a fetch instance.

What do you think? What should the conventions of naming these things be?